Libplanet.Net/Consensus/VoteSet.cs

```diff
@@ -0,0 +1,402 @@
+        }
+
+        internal void AddVerifiedVote(Vote vote)
+        {
+            if (vote.Round != _round ||
+                vote.Flag != _voteType)
+            {
+                throw new InvalidVoteException(
+                    "Round, flag of the vote mismatches",
+                    vote);
+            }
+
+            PublicKey validatorKey = vote.ValidatorPublicKey;
+            BlockHash blockHash = vote.BlockHash;
+
+            Vote? conflicting = null;
+            _logger.Debug("Adding verified vote {Vote}", vote);
+
+            // Already exists in voteSet.votes?
+            if (_votes.ContainsKey(validatorKey))
+            {
+                var existing = _votes[validatorKey];
+                if (existing.BlockHash.Equals(vote.BlockHash))
+                {
+                    throw new InvalidVoteException(
+                        $"{nameof(AddVerifiedVote)}() does not expect duplicate votes",
+                        vote);
+                }
+                else
+                {
+                    conflicting = existing;
+                }
+
+                // Replace vote if blockKey matches voteSet.maj23.
+                if (_maj23 is { } maj23NotNull && maj23NotNull.Equals(blockHash))
+                {
+                    _votes[validatorKey] = vote;
+                }
+
+                // Otherwise don't add it to voteSet.votes
+            }
+            else
+            {
+                // Add to voteSet.votes and incr .sum
+                _votes[validatorKey] = vote;
+            }
+
+            if (_votesByBlock.ContainsKey(blockHash))
+            {
+                if (!(conflicting is null) && !_votesByBlock[blockHash].PeerMaj23)
+                {
+                    // There's a conflict and no peer claims that this block is special.
+                    throw new InvalidVoteException(
+                        "There's a conflict and no peer claims that this block is special",
+                        vote);
+                }
+
+                // We'll add the vote in a bit.
+            }
+            else
+            {
+                // .votesByBlock doesn't exist...
+                if (!(conflicting is null))
+                {
+                    // ... and there's a conflicting vote.
+                    // We're not even tracking this blockKey, so just forget it.
+                    throw new InvalidVoteException("There's a conflicting vote", vote);
+                }
+
+                // ... and there's no conflicting vote.
+                // Start tracking this blockKey
+                _votesByBlock[blockHash] = new BlockVotes(blockHash);
+
+                // We'll add the vote in a bit.
+            }
+
+            BlockVotes votesByBlock = _votesByBlock[blockHash];
+
+            // Before adding to votesByBlock, see if we'll exceed quorum
+            BigInteger origSum = votesByBlock.Sum;
+            BigInteger quorum = _validatorSet.TwoThirdsPower + 1;
```

Comment:
Seems nice, but don't we have to apply same approach to other +2/3 majority?
May be it would be better to `+1` on `ValidatorSet.TwoThirdsPower` etc.
For now, we check `ValidatorSet.TwoThirdsPower < Power to examine`,
but when some decimals were truncated, `ValidatorSet.TwoThirdsPower + 1 <= Power to examine` would be better like this.

Reply:
Do you mean change
```csharp
        public bool HasTwoThirdsAny()
        {
            lock (_lock)
            {
                return Sum > _validatorSet.TwoThirdsPower;
            }
        }
```
to
```csharp
        public bool HasTwoThirdsAny()
        {
            lock (_lock)
            {
                return Sum >= _validatorSet.TwoThirdsPower + 1;
            }
        }
```

Reply:
I think it would be better to modify `ValidatorSet.TwoThirdsPower` to add 1.
I think it's okay to let alone here and change with another PR.